### PR TITLE
ci: add install script

### PR DESCRIPTION
**Pull request by pottekkat**
### Description

Adds an install script and a workflow to publish the script to our s3 bucket. The install script downloads and installs the appropriate ADC binary to make it easy for users to use ADC.

A user should be able to install ADC by just running:

```shell
curl -sL https://run.api7.ai/adc/install | sh
```

Updates to the documentation, both in readme and the docs website will be made after this PR is merged and tested.

Test:

Screenshot 2024-07-22 at 7 20 58 PM

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible

### CODE REVIEW

File: .github/workflows/update-s3-and-cloudfront.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Update S3 and Cloudfront
+on:
+  push:
+    branches:
+      - main
```
Comment (bzp2010) [inline .github/workflows/update-s3-and-cloudfront.yaml:5]:
```suggestion
on:
  push:
    branches:
      - main
    paths:
      - .github/workflows/update-s3-and-cloudfront.yaml
      - s3/install
```

Maybe we could trigger the upload when only two specific files in a push are modified?

File: s3/install [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ARCH="$(uname -m)"
+OS="$(uname)"
+
+# convert to standard arch names used in files
+if [ "x${ARCH}" = "xx86_64" ]; then
+    ARCH="amd64"
+fi
+
+# convert to standard os names used in files
+# TODO: support Windows
```
Comment (bzp2010) [inline s3/install:12]: The bash script is probably hard to run on Windows and should be removed.

**Review by bzp2010 — APPROVED** [review]

File: .github/workflows/update-s3-and-cloudfront.yaml (hunk 1 above)
Comment (pottekkat) [reply .github/workflows/update-s3-and-cloudfront.yaml:5]: Sure. This was similar to how we run this on the docs repo.

File: s3/install (hunk 2 above)
Comment (pottekkat) [reply s3/install:12]: You mean we don't need to care about Windows through this quickstart installer?

Comment (pottekkat) [inline s3/install:12]:
```suggestion
```

**Pull request merged**